recipes/default.rb

```diff
@@ -35,7 +35,7 @@
 end
 
 # If compile_time_update run apt-get update at compile time
-if node['apt']['compile_time_update'] && (!::File.exist?('/var/lib/apt/periodic/update-success-stamp') || !::File.exist?(first_run_file))
+if node['apt']['compile_time_update'] && (!apt_up_to_date? || !::File.exist?(first_run_file))
```

Comment:
I think there may have been some subtle issue where we deliberately skipped the freshness check, which it looks like apt_up_to_date? is adding back in...